tests/ignite/distributed/comp_models/test_native.py

```diff
@@ -35,6 +36,22 @@ def test__native_dist_model():
         _NativeDistModel.create_from_backend("abc")
 
 
+@pytest.mark.distributed
+@pytest.mark.skipif("WORLD_SIZE" in os.environ, reason="Skip if launched as multiproc")
+def test_native_init_method():
```

Comment:
@ahmedo42 are you sure this test is executed ?

Reply:
Yes that was an Issue I faced , and couldn't find  a way around it  , will change it in the new implementation

When running distributed tests using 
`export WORLD_SIZE=2 CUDA_VISIBLE_DEVICES="" pytest --dist=each --tx $WORLD_SIZE*popen//python=python tests/ -m distributed -vvv` it fails and throws the following error : 
 "PyTorch distributed configuration should define env variables" .
 I think it happens because we explicitly set world_size while other env variables are not defined..

Reply:
Yes, to run `export WORLD_SIZE=2 CUDA_VISIBLE_DEVICES="" pytest --dist=each --tx $WORLD_SIZE*popen//python=python tests/ -m distributed -vvv` we have to set up the test as here: https://github.com/pytorch/ignite/blob/e1ce313e7c9fcd6362477f7e2c1b0ce787db5630/tests/ignite/distributed/comp_models/test_native.py#L269-L270

with `local_rank, world_size` fixtures

Reply:
I'll commit here the way I'd like to implement it.
 and let you rework the docs and examples. We can also improve the tests together: should run also on windows and with spawn.

Reply:
Sounds great